File: app/src/main/java/com/github/pockethub/android/ui/item/news/GollumEventItem.java
package com.github.pockethub.android.ui.item.news;

import android.support.annotation.NonNull;
import android.view.View;

import com.github.pockethub.android.ui.StyledText;
import com.github.pockethub.android.ui.view.OcticonTextView;
import com.github.pockethub.android.util.AvatarLoader;
import com.meisolsson.githubsdk.model.GitHubEvent;

public class GollumEventItem extends NewsItem {

    public GollumEventItem(AvatarLoader avatarLoader, GitHubEvent dataItem) {
        super(avatarLoader, dataItem);
    }

    @Override
    public void bind(@NonNull NewsItem.ViewHolder viewHolder, int position) {
        super.bind(viewHolder, position);
        viewHolder.icon.setText(OcticonTextView.ICON_WIKI);

        StyledText main = new StyledText();
        boldActor(main, getData());
        main.append(" updated the wiki in ");
        boldRepo(main, getData());

        viewHolder.event.setText(main);
        viewHolder.details.setVisibility(View.GONE);
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/item/news/GollumEventItem.kt
package com.github.pockethub.android.ui.item.news

import android.view.View
import com.github.pockethub.android.ui.StyledText
import com.github.pockethub.android.ui.view.OcticonTextView
import com.github.pockethub.android.util.AvatarLoader
import com.meisolsson.githubsdk.model.GitHubEvent

class GollumEventItem(avatarLoader: AvatarLoader, dataItem: GitHubEvent) : NewsItem(avatarLoader, dataItem) {

    override fun bind(viewHolder: NewsItem.ViewHolder, position: Int) {
        super.bind(viewHolder, position)
        viewHolder.icon.text = OcticonTextView.ICON_WIKI

        val main = StyledText()
        boldActor(main, data)
        main.append(" updated the wiki in ")
        boldRepo(main, data)

        viewHolder.event.text = main
        viewHolder.details.visibility = View.GONE
    }
}
